owncloudApp/src/main/java/com/owncloud/android/usecases/transfers/uploads/UploadFileFromContentUriUseCase.kt

```diff
@@ -52,13 +58,25 @@ class UploadFileFromContentUriUseCase(
             .build()
 
         val uploadFileFromContentUriWorker = OneTimeWorkRequestBuilder<UploadFileFromContentUriWorker>()
-            .setInputData(inputData)
+            .setInputData(inputDataUploadFileFromContentUriWorker)
             .setConstraints(constraints)
             .addTag(params.accountName)
             .addTag(params.uploadIdInStorageManager.toString())
             .build()
 
-        workManager.enqueue(uploadFileFromContentUriWorker)
+        val behavior = UploadBehavior.fromString(params.behavior)
+        if (behavior == UploadBehavior.MOVE) {
+            val removeLocalFileWorker = OneTimeWorkRequestBuilder<RemoveLocalFileWorker>()
+                .setInputData(inputDataRemoveLocalFileWorker)
+                .build()
+            workManager.beginWith(uploadFileFromContentUriWorker)
+                .then(removeLocalFileWorker) // File is already uploaded, so the original one can be removed if the behaviour is MOVE
+                .enqueue()
+        } else {
+            workManager.beginWith(uploadFileFromContentUriWorker)
+                .enqueue()
```

Comment:
But why `beginWith` and `enqueue` when you know it's going to be just 1 work here? Make it easier, just do it as it was before!

```suggestion
            workManager.enqueue(uploadFileFromContentUriWorker)
```

Reply:
Done